File: monte-carlo-jogo-de-dado.py
import numpy as np
import pandas as pd
import collections as cl
from matplotlib import pylab as plt

# Estimate the prob from the frequency
def obtain_relative_frequency(v):
 count = cl.Counter( v )
 x     = np.array( list( count.keys()   ) )
 fa    = np.array( list( count.values() ) )
 fr    = 1.0*fa/np.sum(fa)
 return x,fr

dado = np.arange(6)
print('dado=',dado) 

# N jogadas de um dado
N       = 10**5
results = np.random.choice(dado, size=N)
vx, vfr = obtain_relative_frequency( results )

# Plot with matplotlib 
plt.bar(dado, vfr)
plt.axhline(y=1/6, color='k', linestyle='--') 
plt.ylabel('Relative Frequency', fontsize=14) 
plt.title('Python: Monte Carlo simulation, N=%d'%(N))
plt.savefig( 'dado-N%d-from-py.png'%(N) )
plt.close()

File: monte-carlo-jogo-de-dado.R
# Estimate the prob from the frequency
obtain_relative_frequency = function(v) {
 df  = data.frame( table(v) ) 
 vx  = as.integer( as.vector(df$v) ) # ~ unique()
 fa  = as.integer( as.vector(df$Freq) ) 
 vfr = fa/sum(fa)  #table(vx)/length(vx) 
 return( data.frame(vx,vfr) )
}


dado = 1:6
print( c('dado=',dado) )


# N jogadas de uma dado
N       = 10^5
results = sample(dado,N,replace=TRUE) 
output  = obtain_relative_frequency( results ) 


png( sprintf('dado-N%d-from-R.png',N) ) 
title=sprintf('R: Monte Carlo simulation, N=%d',N)
xlabel=dado
barplot(output$vfr, main=title,
  names.arg=xlabel, ylab="Relative Frequency")
abline(h=1/length(dado),lty=2)
dev.off() # close


